Added new module: random_wallpaper

rchaput opened the pull request:
I created a new module, named random_wallpaper. This module allows you to set a wallpaper using feh, each time you start py3status. You can also change the wallpaper by clicking on the module in the i3bar.

I've tried to design this module with extensibility and configuration as primary objectives, so you can filter the recognized extensions, select the searched folders,...

tobes commented:
@rchaput thanks for this.  I really like the idea behind this module it's a nice bit of fun.

I think we'll need to do a bit of work to clean things up,  I'm not sure making wrappers for `Py3.log()` etc adds much and also by convention private methods/vars start with a single underscore not two.

also we avoid using `subprocess` where we can in favour of the Py3 helpers.  You shouldn't be using `__init__()` etc.  But these are all things we can fixup.  The main thing is to simplify stuff as much as we can and reduce the amount of code we need to read/maintain.

lasers commented:
Cool. Maybe we shouldn't hardcode this to `feh` as others can do the same thing. See if we can support other commands/arguments too... `feh`, `nitrogen`, `qiv`, `sxiv`, `gpicview`, `imv`, `meh`, et cetera.

tobes commented:
> Maybe we shouldn't hardcode this to feh

My preference would be to get this working with one command and then get it to learn about new ones as we go along.  So we start simple and improve.  Also it really helps with the testing

lasers commented:
Hi @rchaput, it's been a month. Are you planning on making changes anytime soon?

rchaput commented:
Hi there,
Sorry I've been doing some work on other projects.
You can change my code if you want so, otherwise I'll try to clean up the code like you said, maybe this week-end or next week

lasers commented:
Don't worry about it. I was just wondering if you're still around. Good to know. Also, maybe I could do that one day when we get the other things cleared out. Take your time. 👍

lasers commented on py3status/modules/random_wallpaper.py:
```diff
@@ -0,0 +1,302 @@
+                dirs.clear()
+            wallpapers.extend(
+                [os.path.join(root, x) for x in files
+                 if self.__should_add_file(wallpapers, x)])
```
Is this a bug? We're extending the wallpapers... and not returning anything... or am I not understanding something here?

rchaput replied:
What do you mean by returning? For the method' scope or the call to `extend`?
If you talk about `extend`, this is not a bug, `extend` returns `None` and modifies the variable `wallpapers`. Unless there is something I am not aware of (maybe a change between python2 and 3?). It works on my machine, using python3.
If you talk about the whole function, I just chose not to return anything.

lasers replied:
Okay. I think I understand what you did there. You're modifying `wallpapers` via the method `__find_wallpapers` instead of `__find_wallpapers_in_dir` hence no return. I got it now.

As for changes between py2/py3, I'm not seeing any issues. IIRC `list()` instead of `[]` might lead to some unicode bugs related to filenames so it could be something to sneeze about. Not sure.

Still digging. Thanks for the help. 👍

ultrabug commented:
Closing since this is old and idle, re-open if you feel bad about it please

Thanks again

rchaput commented:
Hi there!

Sorry for the delay (2 years already!). I finally had time to delve into this and try to take your comments into account.

Basically I removed some useless code, tried to follow your conventions and I changed the configuration so that users can choose nitrogen, etc. over feh. (but feh is still the default)

I've seen that you modified py3status so that we can deploy our modules as separate packages, so I created a new project and just did that, instead of fiddling with this fork.
The project is available on [my GitHub](https://github.com/rchaput/py3status-random-wallpaper/), you can download a Wheel package in the releases page. I'll gladly take any further comments/suggestions that you may have!
Thank you for the previous comments
